Update tests to send well-formed GUID, better compare JSON responses, fix escaped URLs in tests

aeijdenberg opened the pull request:
This PR should fix the broken tests from yesterday - sorry about that.

(is there a way to set CI up to test PRs before merging?)

jcscottiii commented on helpers/testhelpers/testhelpers.go:
```diff
@@ -286,10 +287,39 @@ func PrepareExternalServerCall(t *testing.T, c *controllers.SecureContext, testS
+func comparePossiblyJSONStrings(a, b string) bool {
+	fallback := (a == b)
+
+	var oa, ob interface{}
```
we can use testify's `JSONEq` https://github.com/stretchr/testify/blob/890a5c3458b43e6104ff5da8dfa139d013d77544/assert/assertions.go#L1128

jcscottiii commented:
> (is there a way to set CI up to test PRs before merging?)

whoops. simple circleci mis-configuration. enabled now

aeijdenberg commented:
That was a whole lot harder than it should have been. :)  It wasn't possibly to simply drop-in the `assert.JSONEq()` as it would fail when we weren't comparing JSON.

Refactored to work, and rebased on current `master`.
